### PR TITLE
Move timezone utils into corehq

### PR DESCRIPTION
**Pull request by dannyroberts**
This is really just
- move dimagi.utils.timezones to corehq.util.timezones: https://github.com/dimagi/commcare-hq/commit/ff70015b340c429a55dab649d16fa46f446cca69
- cleanup https://github.com/dimagi/commcare-hq/commit/f471479a8de04347b107210f73c846db8433d193

but because I did the first one as a subtree merge (of just the dimagi/utils/timezones directory), it pulls in all of dimagi-utils' commits (even ones that seem irrelevant) into the history).

I'm doing this to make it easier to move all timezone-related utility functions to the same place, so that the timezone changes are easier to coordinate.

### CODE REVIEW

File: corehq/util/timezones/fields.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return super(TimeZoneField, self).validate(smart_str(value), model_instance)
+
+    def run_validators(self, value):
+        # coerce value back to a string to validate correctly
+        return super(TimeZoneField, self).run_validators(smart_str(value))
+
+    def to_python(self, value):
+        value = super(TimeZoneField, self).to_python(value)
+        if value is None:
+            return None # null=True
```
Comment (dimagimon) [inline corehq/util/timezones/fields.py]: E261 at least two spaces before inline comment

File: corehq/util/timezones/forms.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from django import forms
+from corehq.util.timezones import zones
+from corehq.util.timezones.utils import coerce_timezone_value
+
+
+class TimeZoneChoiceField(forms.TypedChoiceField):
+    def __init__(self, *args, **kwargs):
+        if not "choices" in kwargs:
```
Comment (dimagimon) [inline corehq/util/timezones/forms.py]: E713 test for membership should be 'not in'

**Comment by czue** [comment]
seems fine, but curious if you considered moving all of dimagi-utils in?

**Comment by dannyroberts** [comment]
It occurred to me but I quickly decided it was a larger project—because other submodules still depend on dimagi utils. However, no other submodules depend on dimagi.utils.timezones, so this seemed like an easy win.

**Comment by czue** [comment]
makes sense, thanks

File: corehq/util/timezones/forms.py (hunk 2 above)
Comment (benrudolph) [inline corehq/util/timezones/forms.py]: wow that's pretty cool that dimagimon caught this

**Comment by esoergel** [comment]
Looks g2m after build.

File: corehq/util/timezones/forms.py (hunk 2 above)
Comment (benrudolph) [inline corehq/util/timezones/forms.py]: didn't realize this was equivalent

**Comment by czue** [comment]
@dannyroberts presumably there's also a deletion from dimagi-utils that needs to happen?

**Comment by dannyroberts** [comment]
yup, thanks for reminding me to PR it

**Comment by dannyroberts** [comment]
https://github.com/dimagi/dimagi-utils/pull/254

**Pull request merged**